.github/workflows/backports.yml

```diff
@@ -0,0 +1,142 @@
+---
+name: mergeit-backport
+
+env:
+  # These should be updated as part of the MAJOR release process
+  # These can be a comma seperated list if we're actively maintaining more releases
+  BACKPORT_CURRENT: 'backport-9'
+  BACKPORT_BUGFIX: 'backport-8'
+  BACKPORT_FORBIDDEN: 'do_not_backport'
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+on:
+  pull_request:
+    types:
+      - labeled
+      - opened
+      - synchronize
+    branches:
+      - main
+
+jobs:
+  changelog-types:
+    # We always skip if do_not_backport has previously been applied.
+    # Otherwise, if someone applies 'mergeit', opens the PR, or pushes a new commit
+    # we'll examine the contents of changelog fragments to try to guess the best backport strategy.
+    if: ${{
+        ! contains(github.event.pull_request.labels.*.name, 'do_not_backport')
+        && (
+          (github.event.action == 'labeled' && github.event.label.name == 'mergeit')
+          || (github.event.action == 'synchronize')
+          || (github.event.action == 'opened')
+        )
+      }}
+
+    permissions:
+      pull-requests: read
+    runs-on: ubuntu-latest
+    outputs:
+      no_backport: ${{ steps.fetch.outputs.no_backport }}
+      bugfix: ${{ steps.fetch.outputs.bugfix }}
+      minor_only: ${{ steps.fetch.outputs.minor_only }}
+    steps:
+      - env:
+          EVENT_CONTEXT: ${{ toJson(github.event) }}
+        run: |
+          echo  "${EVENT_CONTEXT}"
+      - uses: actions/checkout@v2
+      - name: Fetch change types from changelog fragments
+        id: fetch
+        shell: bash {0}
+        run: |
+          gh pr -R "${GITHUB_REPOSITORY}" diff "${{ github.event.pull_request.number }}" --name-only | \
+            grep -E '^changelogs/fragments/' | \
```

Comment:
What about doc changes since they do not include changelog fragments?

Reply:
It'll default to treating them as "minor" changes and backports them to "stable-9"